src/contracts/core/AllocatorManager.sol

```diff
@@ -0,0 +1,965 @@
+     * @dev Note that the `metadataURI` is *never stored * and is only emitted in the `AllocatorMetadataURIUpdated` event
+     * @dev reverts if the caller is delegated to a pre-SDA operator
+     */
+    function registerAsAllocator(
+        address delegationApprover,
+        string calldata metadataURI
+    ) external{
+        require(delegatedTo(msg.sender) == address(0), "DelegationManager.registerAsAllocator: caller is an operator or a staker");
+        _allocatorDetails[msg.sender] = AllocatorDetails({
+            delegationApprover: delegationApprover,
+            isAllocator: true
+        });
+        _setDelegationApprover(msg.sender, delegationApprover);
+        // TODO: events
+    }
+
+    /**
+     * @notice Called by an allocator to emit an `AllocatorMetadataURIUpdated` event indicating the information has updated.
+     * @param metadataURI The URI for metadata associated with an allocator
+     * @dev Note that the `metadataURI` is *never stored * and is only emitted in the `AllocatorMetadataURIUpdated` event
+     */
+    function updateAllocatorMetadataURI(string calldata metadataURI) external {
+        require(isAllocator(msg.sender), "DelegationManager.updateOperatorMetadataURI: caller must be an operator");
+        emit AllocatorMetadataURIUpdated(msg.sender, metadataURI);
+    }
+
+	/**
+	 * @notice Updates the delegation approver for the calling allocator.
+	 *
+	 * @param delegationApprover is the address that must approve of the delegations of a staker's stake to the allocator. If set to 0, no approval is
+	 * required.
+	 */
+    function setDelegationApprover(address delegationApprover) external {
+        _setDelegationApprover(msg.sender, delegationApprover);
+    }
+
+    function _setDelegationApprover(address allocator, address delegationApprover) internal {
+        require(isAllocator(allocator), "DelegationManager._setDelegationApprover: allocator is not registered in EigenLayer");
+        _allocatorDetails[allocator].delegationApprover = delegationApprover;
+        // TODO: events
+    }
+
+    /**
+	 * @notice Completes a handoff queued via queueHandoff.
+	 *
+	 * @param operator the operator in the queued handoff
+	 * @param allocator the allocator in the queued handoff
+	 * @param strategies the strategies to be handed off
+	 * 
+	 * @dev must be called 14 days after the handoff was queued
+	 * @dev the allocator's shares are incremented by the operator's shares for each strategy and the operator's shares are decremented by the operator's
+	 * shares for each strategy.
+	 * @dev if all strategies are not handed off, this function can be called by anyone else to 
+	 * complete the handoff for different strategies
+	 */
+	function completeAllocatorHandoff(address operator, address allocator, IStrategy[] calldata strategies) external {
+        uint256[] memory shares = delegationManager.getMigratableOperatorShares(operator, allocator, strategies);
```

Comment:
does this function also zero out the operator's shares? otherwise I'd be worried this could be called multiple times

Reply:
it does!